src/components/QrReader/qrReader.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import dynamic from "next/dynamic";
+import PropTypes from "prop-types";
+import { getLogger } from "../../utils/logger";
+
+// Need to dynamically import react-qr-reader as it access window directly
+const QrReader = dynamic(import("react-qr-reader"), { ssr: false });
+
+const { error } = getLogger("services:qr");
+
+const QrReaderZone = ({ handleQrScanned }) => {
+  const onScan = data => {
+    if (data) handleQrScanned(data);
+  };
+
+  return (
+    <QrReader
+      data-id="qr-code-reader"
+      delay={100}
+      onError={error}
```

Comment:
No handling of the error ? log the error with text also.

Reply:
The error handling uses the logger under the namespace `services:qr` already. See L9

Reply:
we can put the what is the error also ? currently we are logging "services:qr" text only. we dont know the error.

Reply:
That method logs the ouput, see debug's implm: https://www.npmjs.com/package/debug

Reply:
ahh..okay.